src/wallet/scriptpubkeyman.cpp

```diff
@@ -2652,6 +2652,18 @@ const std::unordered_set<CScript, SaltedSipHasher> DescriptorScriptPubKeyMan::Ge
     return script_pub_keys;
 }
 
+const std::vector<CScript> DescriptorScriptPubKeyMan::GetScriptPubKeysWithMinIndex(int32_t minimum_index) const
+{
+    LOCK(cs_desc_man);
+    std::vector<CScript> script_pub_keys;
+    script_pub_keys.reserve(m_map_script_pub_keys.size());
```

Comment:
d0114e02cddd9bf1cc8760ebe95941f80609259b Because of the `index >= minimum_index` guard (just below, line 2662), could this reserve many more entries than needed? I would probably let `std::vector` increase the capacity as needed.

Reply:
Good catch! We actually know how large the resulting vector should be (`m_map_script_pub_keys.size()` minus the passed minimum index); changed that accordingly and clamped to a minimum of zero to avoid potential integer underflow issues (if e.g. the caller would pass a minimum index higher than the current end range).